### PR TITLE
Add command-line option to control crash-on-fail mode

### PR DESCRIPTION
**Pull request by CiderMan**

### CODE REVIEW

File: include/CppUTest/Utest.h [hunk 1]
```diff
@@ -171,6 +171,10 @@ class UtestShell
 
     virtual SimpleString getMacroName() const;
     TestResult *getTestResult();
+
+    static const NormalTestTerminator normalTestTerminator_;
+    static const CrashingTestTerminator crashingTestTerminator_;
```
Comment (basvodde) [inline include/CppUTest/Utest.h:176]: Why this change?

File: src/CppUTest/CommandLineTestRunner.cpp [hunk 2]
```diff
@@ -95,6 +95,7 @@ void CommandLineTestRunner::initializeTestRun()
     if (arguments_->isColor()) output_->color();
     if (arguments_->runTestsInSeperateProcess()) registry_->setRunTestsInSeperateProcess();
     if (arguments_->isRunIgnored()) registry_->setRunIgnored();
+    if (arguments_->isCrashingOnFail()) registry_->setCrashOnFail();
```
Comment (basvodde) [inline src/CppUTest/CommandLineTestRunner.cpp:98]: Why do this through the registery and not go directly to UTestShell?

File: include/CppUTest/Utest.h (hunk 1 above)
Comment (CiderMan) [reply include/CppUTest/Utest.h:176]: I moved them from statics in the source file to protected statics in the class to make the testing easier.

File: src/CppUTest/CommandLineTestRunner.cpp (hunk 2 above)
Comment (CiderMan) [reply src/CppUTest/CommandLineTestRunner.cpp:98]: I was just following the pattern for the rest of the arguments. As far as I could see, they all went through the registry so I assumed that was the design intent and did the same for the option that I added.

Comment (basvodde) [reply src/CppUTest/CommandLineTestRunner.cpp:98]: Nah, I suggest to not bother with the registry. It seems it doesn't need to know about crash on failure. Adding that won't create any benefits IMHO.

File: tests/CppUTest/TestRegistryTest.cpp [hunk 3]
```diff
@@ -425,3 +430,16 @@ TEST(TestRegistry, reverseZeroTests)
...
+TEST(TestRegistry, doesNotCrashIfNotSetToCrash)
+{
+    CHECK_FALSE(MockTest::wouldCrash());
+}
+
+TEST(TestRegistry, crashesIfSetToCrash)
+{
+    myRegistry->setCrashOnFail();
+
+    CHECK(MockTest::wouldCrash());
```
Comment (basvodde) [inline tests/CppUTest/TestRegistryTest.cpp:443]: Why not use the "static void setCrashMethod(void (*crashme)());" instead of exposing the test to the inner state?

File: include/CppUTest/Utest.h (hunk 1 above)
Comment (basvodde) [reply include/CppUTest/Utest.h:176]: Yeah, please don't do that ;) I've commented below how to change the test so that this isn't needed.

File: tests/CppUTest/TestRegistryTest.cpp (hunk 3 above)
Comment (CiderMan) [reply tests/CppUTest/TestRegistryTest.cpp:443]: Actually, when I remove the Registry from the chain, I don't need these tests and I can revert the Utest.{cpp,h} changes anyway!

**Comment by basvodde** [comment]
Yup easier, now the CppUMock changes, right?
